File: src/main/java/team/exception/sakura/modules/impl/TestModule.java
package team.exception.sakura.modules.impl;

import team.exception.sakura.modules.AbstractModule;
import team.exception.sakura.modules.Category;

public class TestModule extends AbstractModule {

    private static TestModule INSTANCE = null;

    public static TestModule getInstance() {
        if (INSTANCE == null) {
            INSTANCE = new TestModule();
        }
        return INSTANCE;
    }

    public TestModule() {
        super("TestModule", Category.COMBAT);
    }
}

File: src/main/kotlin/team/exception/sakura/modules/impl/render/TestModule.kt
package team.exception.sakura.modules.impl.render

import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.neoforge.client.event.RenderGuiEvent
import org.lwjgl.glfw.GLFW
import team.exception.sakura.modules.AbstractModule
import team.exception.sakura.modules.Category
import team.exception.sakura.utils.nonnull.runSafe

object TestModule: AbstractModule(
    name = "test-module",
    category = Category.RENDER,
    defaultKey = GLFW.GLFW_KEY_Y
) {

    @SubscribeEvent
    fun onRender2D(event: RenderGuiEvent.Post) = runSafe {
        event.guiGraphics.drawString(mc.font, "Hello, world!", 10, 10, 0xFFFFFFFF.toInt())
    }

}